src/unix/fs.c

```diff
@@ -739,6 +734,36 @@ static int uv__fs_fstat(int fd, uv_stat_t *buf) {
 }
 
 
+typedef ssize_t (*uv__fs_buf_iter_yield)(uv_fs_t* req);
+static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_yield yield) {
+  unsigned int iovmax=uv__getiovmax();
+  unsigned int nbufs=req->nbufs;
+  uv_buf_t *bufs=req->bufs;
+  ssize_t total=0;
+  ssize_t result;
+
+  while(nbufs) {
+    req->nbufs=nbufs;
+    if(req->nbufs>iovmax) req->nbufs=iovmax;
+
+    result=yield(req);
+    if(result<=0) {
+      total=result;
```

Comment:
It's debatable whether this is correct... if `total > 0`, at least some data has been read or written.